tests/compiler/std/operator-overloading.untouched.wat

```diff
@@ -1263,25 +1270,39 @@
       )
      )
     )
-    (set_local $32
-     (select
-      (f64.const 1.5)
-      (f64.const 1)
-      (get_local $11)
+    (set_local $33
+     (block $~lib/math/casted_select<u64,f64>|inlined.0 (result f64)
+      (set_local $32
+       (f64.const 1.5)
+      )
+      (set_local $33
+       (f64.const 1)
+      )
+      (f64.reinterpret/i64
+       (select
+        (i64.reinterpret/f64
+         (get_local $32)
```

Comment:
Is it actually the case that `casted_select` works as intended here? I was under the impression that the reason for the described advantage was that the `i64.reinterpret/f64` was precomputed due to its argument being a constant, but once that becomes a helper function the constant becomes a local and precomputation is, if at all, performed by Binaryen and only when optimizing.

Reply:
Yes, in optimized version (with propageted constants) it would just one reinterpret.

Reply:
Hmm, somehow this looks like an over-optimization to me that we shouldn't do. Maybe Binaryen could do that, so that we can just use normal selects?

Reply:
Sure, may be in future VMs handle this case better than now. Ok I'll switch to normal `select`

Reply:
done!